### PR TITLE
Tpetra: Add unit test for #3968

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra @rppawlo 

This was originally part of PR #3996.  @rppawlo pushed an improved fix to #3968 first in PR #3999, which got merged.  I thus deleted PR #3996. This commit includes just the unit test that was part of PR #3996, but not part of PR #3999.

## Related Issues

* Follows #3999 
* Related to #3968

### CODE REVIEW

**Comment by mhoemmen** [comment]
I just fixed the build issue.  Let's try this again.

File: packages/tpetra/core/test/ImportExport/Issue_3968.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  // least 64 bits, so prefer that over long or unsigned long.
+#if defined(HAVE_TPETRA_INST_INT_LONG_LONG)
+  using GO = long long; // always at least 64 bits
+#elif defined(HAVE_TPETRA_INST_INT_LONG)
+  using GO = long; // may be 64 or 32 bits
+#elif defined(HAVE_TPETRA_INST_INT_UNSIGNED_LONG)
+  using GO = unsigned long; // may be 64 or 32 bits
+#else
+  using GO = Tpetra::Map<>::global_ordinal_type;
+#endif
```
Comment (rppawlo) [inline packages/tpetra/core/test/ImportExport/Issue_3968.cpp]: Does this mean that the Tpetra default favors int over long long? I would have expected long long.

File: packages/tpetra/core/test/ImportExport/Issue_3968.cpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    std::vector<GO> gblInds (lclNumInds);
+    std::iota (std::begin (gblInds), std::end (gblInds), GO (0));
+    const GO indexBase = 0;
+    return Teuchos::rcp (new map_type (flag, gblInds.data (), lclNumInds, indexBase, comm));
+  }
+
+  // "Base value" should be some value, not default, not negative (in
+  // case GO is unsigned, which is not default but possible).  Pick
+  // baseValue() > comm->getSize().
+  constexpr GO baseValue () {
```
Comment (rppawlo) [inline packages/tpetra/core/test/ImportExport/Issue_3968.cpp]: Maybe add:
TEUCHOS_ASSERT(baseValue > comm->getSize());

Comment (mhoemmen) [reply packages/tpetra/core/test/ImportExport/Issue_3968.cpp]: True -- no need for `constexpr` here.

File: packages/tpetra/core/test/ImportExport/Issue_3968.cpp (hunk 1 above)
Comment (mhoemmen) [reply packages/tpetra/core/test/ImportExport/Issue_3968.cpp]: @rppawlo Fun fact: Default GlobalOrdinal type is `int`, even though `long long` is enabled by default.  See https://github.com/trilinos/Trilinos/issues/4029 .

**Comment by mhoemmen** [comment]
Thanks @rppawlo ! :-D

**Comment by mhoemmen** [comment]
Thanks @rppawlo ! :D

**Pull request merged**